js/aax.js

```diff
@@ -2376,29 +2410,48 @@ module.exports = class aax extends Exchange {
     }
 
     parseTransfer (transfer, currency = undefined) {
-        const data = this.safeValue (transfer, 'data', {});
-        const id = this.safeString (data, 'transferID');
-        const dateTime = this.safeString (data, 'transferTime');
-        const timestamp = this.safeNumber (transfer, 'ts');
-        const currencyCode = this.safeString (currency, 'code');
-        const responseCode = this.safeString (transfer, 'code');
-        let status = 'canceled';
-        if (responseCode === '1') {
-            status = 'ok';
+        //     {
+        //          quantity: '0.000010000000',
+        //          transferID: '480975741034369024',
+        //          transferTime: '2022-03-24T13:53:07.042Z',
+        //          fromPurse: 'VLTP',
+        //          toPurse: 'SPTP',
+        //          currency: 'ETH'
+        //     },
+        const id = this.safeString (transfer, 'transferID');
+        const amount = this.safeNumber (transfer, 'quantity');
+        const dateTime = this.safeString (transfer, 'transferTime');
+        const timestamp = this.parse8601 (dateTime);
+        const accounts = this.safeValue (this.options, 'accounts', {});
+        const fromId = this.safeString (transfer, 'fromPurse');
+        const toId = this.safeString (transfer, 'toPurse');
+        const fromAccount = this.safeString (accounts, fromId);
+        const toAccount = this.safeString (accounts, toId);
+        let currencyCode = this.safeString (transfer, 'currency');
+        if (currencyCode === undefined) {
+            currencyCode = this.safeString (currency, 'code');
```

Comment:
Let's rewrite lines 2430-2432 as we always apply the same code with safeCurrencyCode for all currencies:

```JavaScript
const currencyId = this.safeString (transfer, 'currency');
const currencyCode = this.safeCurrencyCode (currencyId, currency);
```

Note, that we propagate the `currency` argument to `safeCurrencyCode` as a fallback.